src/wallet/spend.cpp

```diff
@@ -488,6 +498,23 @@ CoinsResult AvailableCoins(const CWallet& wallet,
         }
     }
 
+    // Return all the coins from one TRUC transaction, that have the highest value.
+    // This could be improved in the future by encoding these restrictions in
+    // the coin selection itself so that we don't have to filter out
+    // other unconfirmed TRUC coins beforehand.
+    if (params.check_version_trucness && unconfirmed_truc_coins.size() > 0) {
+        auto highest_value_truc_tx = std::max_element(truc_txid_by_value.begin(), truc_txid_by_value.end(), [](const auto& tx1, const auto& tx2){
+                return tx1.second < tx2.second;
+                });
+
+        const Txid& truc_txid = highest_value_truc_tx->first;
+        for (const auto& [type, output] : unconfirmed_truc_coins) {
+            if (output.outpoint.hash == truc_txid) {
+                    result.Add(type, output);
```

Comment:
In c5a2d080116270ecd0414c14eb412fa30eaaedaf "wallet: don't return utxos from multiple truc txs in AvailableCoins"

nit: extra indent